elasticsearch/run.sh

```diff
@@ -216,4 +216,8 @@ verify_or_add_index_templates() {
 
 verify_or_add_index_templates &
 
+if [ -v HEAP_DUMP_LOCATION ]; then
+    info Setting heap dump location "$HEAP_DUMP_LOCATION"
+    export JAVA_OPTS="${JAVA_OPTS:-} -XX:HeapDumpPath=$HEAP_DUMP_LOCATION"
```

Comment:
If HeapDumpPath is not set in JAVA_OPTS, will Elasticsearch dump the heap?  If so, then we need to set this _always_, and the default should be `/elasticsearch/persistent` unless that is defined in a variable somewhere.